src/Scientist/Observation.cs

```diff
@@ -1,49 +1,178 @@
-﻿using GitHub.Internals;
+﻿using NullGuard;
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace GitHub
 {
     /// <summary>
-    /// Defines an observation of an experiment's execution.
+    /// Provides an interface for an observation of an experiment's execution.
     /// </summary>
-    public class Observation
+    public interface IObservation
```

Comment:
I built this and `IResult` to encapsulate the generic parameter on their implementations to keep the publishing method non-generic.  Here are a couple of thoughts:

* Update the publishing interface to be `Task Publish<T>(Result<T> result)`
* Provide a mechanism for scoping publishing to the specific type to allow `Task Publish(Result<T> result)` where the generic is defined on the class itself.
* Keep the interface to encapsulate the generic, and the publishing interface the same `Task Publish(IResult result)`.

Let me know what you think; i'm open to any of the avenues.